sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -389,10 +389,12 @@ def validate_composite_type_param(type_param, error_msg_prefix):
   if sys.version_info.major == 3 and sys.version_info.minor >= 10:
     if isinstance(type_param, types.UnionType):
       is_not_type_constraint = False
-  is_forbidden_type = (
-      isinstance(type_param, type) and type_param in DISALLOWED_PRIMITIVE_TYPES)
+  if sys.version_info.major == 3 and sys.version_info.minor < 9:
```

Comment:
Could you add a comment explaining the pre/post 3.9, 3.10 logic in this block?

Reply:
Done